components/script/dom/filereader.rs

```diff
@@ -245,7 +272,28 @@ impl FileReader {
             format!("data:{};base64,{}", data.blobtype, base64)
         };
 
-        DOMString::from(output)
+        StringOrObject::String(DOMString::from(output))
+    }
+
+    // https://w3c.github.io/FileAPI/#dfn-readAsArrayBuffer
+    #[allow(unsafe_code, unused_variables)]
+    fn perform_readasarraybuffer(cx: *mut JSContext, data: ReadMetaData, bytes: &[u8])
+        -> StringOrObject {
+        unsafe {
+            let length = bytes.len() as u32;
+            let newArrayBuffer = JS_NewArrayBuffer(cx, length);
```

Comment:
We need to create a JSAutoCompartment value in this method before calling JS_NewArrayBuffer.